flask_app/api/predictions.py

```diff
@@ -312,8 +310,9 @@ def controller(relation, taxon_id, is_subject_query, conf_thresh, strict, check_
         species_list, taxon_ids = get_species_list_from_phyla(appropriate_phyla)
         missed_taxa = []
     else:
+        check_list = [eval(i) for i in check_list]
         species_list, taxon_ids = get_taxonomy_from_species_list(check_list)
-        missed_taxa = list_diff(check_list, taxon_ids) # this is the list of taxa not in the database
+        missed_taxa = list_diff(check_list, taxon_ids)  # this is the list of taxa not in the database
```

Comment:
Good that you found out why it was not working! If I may, I would recommend to use some other method than **eval**. Eval is considered not the best to make use of: https://stackoverflow.com/questions/1832940/why-is-using-eval-a-bad-practice

Reply:
Did not know that! Thanks for the heads-up :)